### PR TITLE
Syncs a check's output with the catalog when output rate limiting isn't in effect.

### LINKED ISSUE #1057: check Output not updated at all
Hi all,

Has anyone experienced the same situation like below:
1) consul agent is configured with a watcher on critical checks
config:
```
{
  "type": "checks",
  "handler": "/home/imcom/consul/bin/rasalghul watch -name critical -region dc2",
  "datacenter": "dc2",
   "state": "critical"
 }
```
2) deploy a script check by adding a json file in Consul config dir and reload consul agent.

3) script checks output text/json formatted string to stdout and run every 30s or whatever interval.

4) check_update_interval set to "0s" to have output get synced each time check runs.
```
{
   "check_update_interval": "0s",
   ...
}
```

Then, from the input of watcher, I get empty string in `Output` field every time for the first run of checks and then this field sometimes will be never filled even though the checks keep running.

```
{"Node":"rdserver9","CheckID":"service:rdserver9.systembasic:systembasic:1","Name":"Service 'systembasic:systembasic' check","Status":"critical","Notes":"load","Output":"","ServiceID":"rdserver9.systembasic:systembasic","ServiceName":"systembasic:systembasic"}]
```

From watcher's perspective, the checks will be in critical status anyway and have no outputs by the first time watcher receives the checks, so I get critical checks by the time the checks are first loaded with output filed being empty. What's bothering here is that the Output field will not update afterwards, even though the checks keep running ...

I also noted that TTL checks do not have this behavior ... Output returns immediately each time

Any ideas?

### PR DESCRIPTION
**Pull request by slackpad**
Fixes #1057.

### CODE REVIEW

File: command/agent/local.go [hunk 1]
```diff
@@ -499,6 +516,8 @@ func (l *localState) syncChanges() error {
 		if err := l.syncNodeInfo(); err != nil {
 			return err
 		}
+	} else {
+		l.logger.Printf("[DEBUG] agent: Node info in sync")
```
Comment (sean-) [inline command/agent/local.go]: `s/DEBUG/TRACE/` ?

File: command/agent/local.go [hunk 2]
```diff
@@ -434,11 +434,28 @@ func (l *localState) setSyncState() error {
...
 			eCopy := new(structs.HealthCheck)
 			*eCopy = *existing
-			eCopy.Output = ""
-			check.Output = ""
-			equal = eCopy.IsSame(check)
+
+			// Copy the server's check before modifying, otherwise
+			// in-memory RPC-based unit tests will have side effects.
+			cCopy := new(structs.HealthCheck)
+			*cCopy = *check
```
Comment (sean-) [inline command/agent/local.go]: Even though the Copy() method will be just an assignment, can this be moved to a `Copy()` method?  If `HealthCheck` ever picks up something that can't be shallow copied, we won't have to dig around finding all assignments?

**Comment by sean-** [comment]
Small comments, LGTM otherwise.  It'll be really nice to see this issue go away.

File: command/agent/local.go (hunk 1 above)
Comment (slackpad) [inline command/agent/local.go]: This one is low-rate and not really printed in response to traffic so I figured it was `DEBUG`-worthy.

File: command/agent/local.go (hunk 2 above)
Comment (slackpad) [inline command/agent/local.go]: That's a good call, especially since we've centralized `IsSame`.

File: command/agent/local.go (hunk 1 above)
Comment (sean-) [inline command/agent/local.go]: Cool. I was concerned this would fire per check and would be spammy, but it looks like it's called only once per AE run.

**Pull request merged**